fix(ivy): don't throw when attempting to destroy a destroyed ComponentRef

crisbeto opened the pull request:
Currently in Ivy we throw when attempting to destroy a `ComponentRef` that has been destroyed, however in ViewEngine we didn't which can cause some tests to break. These changes remove the error to match ViewEngine.

These changes resolve FW-1379.

pkozlowski-opensource commented on packages/core/test/acceptance/view_container_ref_spec.ts:
```diff
@@ -1061,7 +1098,7 @@ describe('ViewContainerRef', () => {
           <ng-template #foo>
             <div>{{name}}</div>
           </ng-template>
-          
```
nit: curious about those white-spaces in the diff? Do you have a different formatter or this file was badly formatted?

crisbeto replied:
That's just my IDE stripping the trailing whitespace.

pkozlowski-opensource commented on packages/core/src/render3/component_ref.ts:
```diff
@@ -258,13 +257,20 @@ export class ComponentRef<T> extends viewEngine_ComponentRef<T> {
+    if (this.destroyCbs) {
+      this.destroyCbs.push(callback);
+    } else if (ngDevMode) {
+      console.warn('ComponentRef already destroyed');
```
Thinking about this:
* we are duplicating code (warning message) - it is one line, but still...
* VE was not doing this, so... 
* we are suing `console.warn` directly - not sure if we report warning like this in other places

I don't know, I guess I'm trying to say that maybe this check + warning is just "not worth it"? 
I could definitely live without this warning... WDYT?

crisbeto replied:
The duplicated code should be stripped in production mode. I decided to leave it in as a warning, because I was assuming that whoever put the original error in had a reason to do it. That being said, I can live without the warning as well.

crisbeto commented:
@pkozlowski-opensource the warning has been removed.

crisbeto committed:
fix(ivy): don't throw when attempting to destroy a destroyed ComponentRef

Currently in Ivy we throw when attempting to destroy a `ComponentRef` that has been destroyed, however in ViewEngine we didn't which can cause some tests to break. These changes remove the error to match ViewEngine.

These changes resolve FW-1379.

pkozlowski-opensource reviewed (approved):
LGTM

AndrewKushnir commented:
- [VE Presubmit](https://test.corp.google.com/ui#id=OCL:253065092:BASE:253065147:1560448952560:2f662b)
- [Ivy Presubmit](https://test.corp.google.com/ui#id=OCL:253065816:BASE:253066111:1560449171235:c87516c0)

kara reviewed (approved):
LGTM

AndrewKushnir commented:
FYI: VE presubmit is successful, Ivy presubmit looks normal. Thank you.
